control_plane/src/storage.rs

```diff
@@ -96,46 +96,58 @@ impl PageServerNode {
             .unwrap()
     }
 
-    pub fn init(&self, create_tenant: Option<&str>) -> anyhow::Result<()> {
+    pub fn init(
+        &self,
+        create_tenant: Option<&str>,
+        config_overrides: &[&str],
+    ) -> anyhow::Result<()> {
         let mut cmd = Command::new(self.env.pageserver_bin()?);
         let var = "LLVM_PROFILE_FILE";
         if let Some(val) = std::env::var_os(var) {
             cmd.env(var, val);
         }
 
         // FIXME: the paths should be shell-escaped to handle paths with spaces, quotas etc.
+        let base_data_dir_param = self.env.base_data_dir.display().to_string();
+        let pg_distrib_dir_param =
+            format!("pg_distrib_dir='{}'", self.env.pg_distrib_dir.display());
+        let authg_type_param = format!("auth_type='{}'", self.env.pageserver.auth_type);
+        let listen_http_addr_param = format!(
+            "listen_http_addr='{}'",
+            self.env.pageserver.listen_http_addr
+        );
+        let listen_pg_addr_param =
+            format!("listen_pg_addr='{}'", self.env.pageserver.listen_pg_addr);
         let mut args = vec![
-            "--init".to_string(),
-            "-D".to_string(),
-            self.env.base_data_dir.display().to_string(),
-            "-c".to_string(),
-            format!("pg_distrib_dir='{}'", self.env.pg_distrib_dir.display()),
-            "-c".to_string(),
-            format!("auth_type='{}'", self.env.pageserver.auth_type),
-            "-c".to_string(),
-            format!(
-                "listen_http_addr='{}'",
-                self.env.pageserver.listen_http_addr
-            ),
-            "-c".to_string(),
-            format!("listen_pg_addr='{}'", self.env.pageserver.listen_pg_addr),
+            "--init",
+            "-D",
+            &base_data_dir_param,
+            "-c",
+            &pg_distrib_dir_param,
+            "-c",
+            &authg_type_param,
+            "-c",
+            &listen_http_addr_param,
+            "-c",
+            &listen_pg_addr_param,
```

Comment:
I'd suggest structuring this like:
```
let mut args = Vec:new();
args.extend(["-D", self.env.base_data_dir.display().to_string()]);
args.extend(["-c", format!("auth_type='{}'", self.env.pageserver.auth_type)]);
...
```
That makes it slightly easier to add or remove options